src/core/toga/interface/widgets/switch.py

```diff
@@ -0,0 +1,102 @@
+from .base import Widget
+
+
+class Switch(Widget):
+    """
+    Switch widget, a clickable button with two stable states, True (on, checked) and False (off, unchecked)
+
+    :param label:       Text to be shown next to the Switch
+    :type label:        ``str``
+
+    :param id:          An identifier for this widget.
+    :type  id:          ``str``
+
+    :param style:       an optional style object. If no style is provided then a
+                        new one will be created for the widget.
+    :type style:        :class:`colosseum.CSSNode`
+
+    :param on_press:    Function to execute when pressed
+    :type on_press:     ``callable``
+
+    :param state        Current state of the Switch
+    :type state         ``Bool`
+    """
+
+    def __init__(self, label, id=None, style=None, on_press=None, state=False):
+        super().__init__(id=id, style=style, label=label, on_press=on_press, state=state)
+
+    def _configure(self, label, on_press, state):
+        self.label = label
+        self.on_press = on_press
+        self.state = state
```

Comment:
@freakboy3742 I agree, there should be an ```widget.enabled```. I'm going to work on that.

I don't know if there is any convention against two worded properties but if not we could do it iOS style. 
iOS uses ```is​On``` instead of just ```on```, derived from that we could use ```switch.is_on```. Then it would work nicely with ```if switch.is_on``` and also ```switch.is_on = True```.